src/storage/txn/commands.rs

```diff
@@ -270,17 +273,30 @@ pub trait CommandExt {
     fn gen_lock(&self, _latches: &Latches) -> latch::Lock;
 }
 
+macro_rules! ctx {
```

Comment:
Does this need to be a separate macro? Could it be inlined into command! ?

Reply:
> Does this need to be a separate macro? Could it be inlined into `command!` ?

I'm afraid it could not. I can't find a way to put things in `CommandExt` into `command!` because for some certain commands, it has to provide a manual implemention for some methods in `CommandExt`, and override these methods inside macro is really hard because the problem of `self` and Hygienic macro. And we cannot `impl CommandExt` multiple times.
However, I found that most of the usage of `ctx` are in `process_read_impl` and `process_write_impl`, with some furtuer refactor (might be related to [the discussion in sig-trasaction](https://github.com/tikv/sig-transaction/pull/28/)) probabliy we'll not need `ctx` at all in the future.